pkg/nsx/services/securitypolicy/store.go

```diff
@@ -1,153 +1,110 @@
 package securitypolicy
 
 import (
-	"fmt"
-	"strconv"
-	"strings"
-	"sync"
-
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
 	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"k8s.io/client-go/tools/cache"
 
 	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
-	nsxutil "github.com/vmware-tanzu/nsx-operator/pkg/nsx/util"
-	"github.com/vmware-tanzu/nsx-operator/pkg/util"
 )
 
-func queryTagCondition(service *SecurityPolicyService) string {
-	return fmt.Sprintf("tags.scope:%s AND tags.tag:%s",
-		strings.Replace(util.TagScopeCluster, "/", "\\/", -1),
-		strings.Replace(service.NSXClient.NsxConfig.Cluster, ":", "\\:", -1))
+type resourceAssertion func(i interface{}) interface{}
+
+func keyFunc(obj interface{}) (string, error) {
+	switch v := obj.(type) {
+	case model.Group:
+		return *v.Id, nil
+	case model.SecurityPolicy:
+		return *v.Id, nil
+	case model.Rule:
+		return *v.Id, nil
+	}
+	return "", nil
 }
 
-func queryGroup(service *SecurityPolicyService, wg *sync.WaitGroup, fatalErrors chan error) {
-	defer wg.Done()
-	queryParam := fmt.Sprintf("%s:%s", common.ResourceType, common.ResourceTypeGroup) + " AND " + queryTagCondition(service)
-	var cursor *string = nil
-	pageSize := common.PageSize
-	for {
-		response, err := service.NSXClient.QueryClient.List(queryParam, cursor, nil, &pageSize, nil, nil)
-		err = common.TransError(err)
-		if _, ok := err.(nsxutil.PageMaxError); ok == true {
-			common.DecrementPageSize(&pageSize)
-			continue
-		}
-		if err != nil {
-			fatalErrors <- err
-		}
-		for _, g := range response.Results {
-			a, err := Converter.ConvertToGolang(g, model.GroupBindingType())
-			if err != nil {
-				for _, e := range err {
-					fatalErrors <- e
-				}
-			}
-			c, _ := a.(model.Group)
-			err2 := service.GroupStore.Add(c)
-			if err2 != nil {
-				fatalErrors <- err2
-			}
-		}
-		cursor = response.Cursor
-		if cursor == nil {
-			break
-		}
-		c, _ := strconv.Atoi(*cursor)
-		if int64(c) >= *response.ResultCount {
-			break
-		}
+func indexFunc(obj interface{}) ([]string, error) {
```

Comment:
Pleas add some comments for this indexFunc as well

Reply:
ok